tools/lsp/completion.rs

```diff
@@ -318,6 +320,16 @@ pub(crate) fn completion_at(
         })
         .collect();
         return Some(r);
+    } else if node.kind() == SyntaxKind::State {
+        let r: Vec<_> = [("when", "when $1: {\n    $0\n}")]
```

Comment:
Did you check that in only recommand that after the state name?
Can you write a test for that?